### PR TITLE
logger.warn is deprecated, use logger.warning instead

### PR DESCRIPTION
**Pull request by rahulporuri**
Ref : https://docs.python.org/3.6/library/logging.html#logging.Logger.warning

### CODE REVIEW

**Comment by rahulporuri** [comment]
```
test_set_object_data (pyface.tests.test_clipboard.TestClipboard) ... Command "coverage run -p -m unittest discover -v pyface" exited with segmentation fault (-11)

Command '['/Users/travis/.edm/envs/edm/bin/edm', 'run', '-e', 'pyface-test-3.6-wx', '--', 'coverage', 'run', '-p', '-m', 'unittest', 'discover', '-v', 'pyface']' returned non-zero exit status 245.
```

[log9.txt](https://github.com/enthought/pyface/files/5136765/log9.txt)

**Comment by corranwebster** [comment]
We're getting the clipboard segfault for wx on MacOS now?

File: docs/source/conf.py [hunk 1]
```diff
@@ -150,7 +150,7 @@
 except ImportError as exc:
     import warnings
     msg = "Can't find Enthought Sphinx Theme, using default.\nException was:\n{}"
-    warnings.warn(RuntimeWarning(msg.format(exc)))
+    warnings.warning(RuntimeWarning(msg.format(exc)))
```
Comment (kitchoi) [inline docs/source/conf.py:153]: `warnings.warning` is not defined.

**Review by kitchoi — APPROVED** [review]
LGTM

**Pull request merged**